protobuf-codegen-identical-test/tests/diff.rs

```diff
@@ -82,6 +89,97 @@ fn print_diff(dir: &Path, a: &Path, b: &Path) {
     print!("{}", str::from_utf8(&output.stderr).unwrap());
 }
 
+fn protoc_descriptor_set(includes: &[PathBuf], inputs: &[PathBuf]) -> FileDescriptorSet {
+    let mut temp_file = NamedTempFile::new().unwrap();
+    protoc::Protoc::from_env_path()
+        .descriptor_set_out_args()
+        .out(temp_file.path())
+        .includes(includes)
+        .inputs(inputs)
+        .write_descriptor_set()
+        .unwrap();
+    protobuf::parse_from_reader(&mut temp_file).unwrap()
+}
+
+fn pure_descriptor_set(includes: &[PathBuf], inputs: &[PathBuf]) -> FileDescriptorSet {
+    let mut codegen = protobuf_codegen_pure::parse_and_typecheck(includes, inputs).unwrap();
+    let relative_paths: HashSet<_> = codegen
+        .relative_paths
+        .iter()
+        .map(|path| path.to_string_lossy())
+        .collect();
+    codegen
+        .file_descriptors
+        .retain(|fd| relative_paths.contains(fd.get_name()));
+    let mut fds = FileDescriptorSet::new();
+    fds.file = RepeatedField::from_vec(codegen.file_descriptors);
+    fds
+}
+
+fn normalize_descriptor_set(fds: &mut FileDescriptorSet) {
+    for desc in fds.file.iter_mut() {
+        normalize_file_descriptor(desc)
+    }
+}
+
+fn normalize_file_descriptor(desc: &mut FileDescriptorProto) {
+    if !desc.has_syntax() {
+        desc.set_syntax("proto2".into())
```

Comment:
Could be

```
desc.set_syntax(FileDescriptorProto::default_instance().get_syntax())
```

Up to you.

Reply:
Actually, that seems to return `""`, so leaving as is.